actors.md

```diff
@@ -688,26 +677,25 @@ func SubmitPost(proofs PoStProof, faults []FaultSet, recovered Bitfield, done Bi
 		TransferFunds(msg.From, msg.Value-feesRequired)
 	}
 
-	if !CheckPostProof(miner.SectorSize, proof, faults) {
+    seed := GetRandFromBlock(self.ProvingPeriodStart + POST_CHALLENGE_TIME)
+    faultSet := self.faultSet
+
+	if !VerifyPost(self.SectorSize, self.provingSet, seed, proof, faultSet) {
 		Fatal("proof invalid")
 	}
 
-	// combine all the fault set bitfields, and subtract out the recovered
-	// ones to get the set of sectors permanently lost
-	permLostSet = allFaults.Subtract(recovered)
+    // clear fault set
+    self.faultSet = EmptySectorSet()
 
-	// burn funds for fees and collateral penalization
-	self.BurnFunds(CollateralForSize(self.SectorSize*permLostSet.Size()) + feesRequired)
+    // TODO: penalize for faults
 
-	// update sector sets and proving set
-	miner.Sectors.Subtract(done)
-	miner.Sectors.Subtract(permLostSet)
+	// Remove doneSet from the current sectors
+	miner.Sectors.Subtract(doneSet)
 
-	// update miner power to the amount of data actually proved during
-	// the last proving period.
+	// Update miner power to the amount of data actually proved during the last proving period.
 	oldPower := miner.Power
 
-	miner.Power = (miner.ProvingSet.Size() - allFaults.Count()) * miner.SectorSize
+	miner.Power = (miner.ProvingSet.Size() - faultSet.Count()) * miner.SectorSize
```

Comment:
You just `// clear fault set` above, so the `- faultSet.Count()` isn't going to do much.

Reply:
faultSet is a local copy, I clear `self.faultSet`

Reply:
Ah hah. Thanks.

Reply:
IMO the state updates could be re-arranged into a clearer block at the end of the method:
```
// Update miner power to the amount of data actually proved during the last proving period.
self.Power = (self.ProvingSet.Size() - self.FaultSet.Count()) * self.SectorSize

// Done sectors no longer need to be proven
self.Sectors.Subtract(doneSet)
self.ProvingSet = self.Sectors

// Faults are assumed temporary unless the miner has also declared the sector "done"
self.FaultSet = EmptySectorSet()

... // deal state
```